### PR TITLE
Use null propagation

### PR DESCRIPTION
**Pull request by drewnoakes**
## Proposed changes

- Use null propagation

## Test methodology 

- Manual review. It compiles. CI will run tests.

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: ResourceManager/Xliff/TranslationUtil.cs [hunk 1]
```diff
@@ -197,12 +197,7 @@ public static void TranslateItemsFromList(string category, ITranslation translat
...
         {
-            if (obj is null)
-            {
-                return;
-            }
-
-            var property = obj.GetType().GetProperty(propName, _propertyFlags);
+            var property = obj?.GetType().GetProperty(propName, _propertyFlags);
 
             if (property is null)
```
Comment (ivangrek) [inline ResourceManager/Xliff/TranslationUtil.cs:202]: It looks like a case.

Comment (drewnoakes) [reply ResourceManager/Xliff/TranslationUtil.cs:202]: Do you mean a switch? I'm not sure what you mean exactly, sorry.

Comment (ivangrek) [reply ResourceManager/Xliff/TranslationUtil.cs:202]: Designated case: "Use null propagation"

Comment (ivangrek) [reply ResourceManager/Xliff/TranslationUtil.cs:202]: `property?.SetValue(obj, value, null);`

Comment (drewnoakes) [reply ResourceManager/Xliff/TranslationUtil.cs:202]: Thanks, I see what you mean now.

**Review by gerhardol — APPROVED** [review]
OK, just agree with ivangrek

File: ResourceManager/Xliff/TranslationUtil.cs (hunk 1 above)
Comment (ivangrek) [reply ResourceManager/Xliff/TranslationUtil.cs:202]: I found this option. But... 😄 

```
string value = translation.TranslateItem(category, propName, "Text", ProvideDefaultValue);

if (!string.IsNullOrEmpty(value))
{
    return;
}

var property = obj?.GetType()
    .GetProperty(propName, _propertyFlags);

if (property?.CanWrite is true)
{
    property.SetValue(obj, value, null);
}
```

Comment (drewnoakes) [reply ResourceManager/Xliff/TranslationUtil.cs:202]: I'm not sure whether it's worth changing this. The current behaviour bails early when property is null, avoiding redundant work. Feel free to push to this branch if you'd like to see a change. For now, I can't see one that I like.

**Review by ivangrek — APPROVED** [review]

**Pull request merged**